lib/intercepted_request_router.js

```diff
@@ -40,15 +40,14 @@ class InterceptedRequestRouter {
 
     this.response = new IncomingMessage(this.socket)
     this.playbackStarted = false
+    this.playbackOnSocket = false
```

Comment:
Would calling this `waitingForSocketEvent` make this a little clearer?

Either way, it could be useful to add a couple lines of documentation about what these flags mean.

Reply:
Naming variables is the hardest part of coding for me.
I can rename it is that's more clear.
There are comments around this flag down where it's set. I had a bunch of comments/notes in the file from my research and I tried to declutter it.  So I need refresh eyes to let me know what doesn't make sense.
Should I move the comments lines 253-254 here or copy them?

Reply:
Since this is the constructor, where it's first set, it seems like a great place to put comments. I think the other comment makes sense where it is, so I'd suggest copying it, and then expanding it a bit to explain the lifecycle of this flag.

Maybe something like this:

> For parity with Node, it's important the socket event is emitted before we begin playback. This flag is set when playback is triggered if we haven't yet gotten the socket event to indicate that playback should start as soon as it comes in.

That description makes me think it should be called `readyToStartPlaybackOnSocketEvent`.